exporter/coralogixexporter/metrics_client.go

```diff
@@ -0,0 +1,164 @@
+package coralogixexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+func newMetricsExporter(cfg config.Exporter, set component.ExporterCreateSettings) (*exporter, error) {
+	oCfg := cfg.(*Config)
+
+	if oCfg.Metrics.Endpoint == "" || oCfg.Metrics.Endpoint == "https://" || oCfg.Metrics.Endpoint == "http://" {
+		return nil, errors.New("coralogix exporter config requires `metrics.endpoint` configuration")
+	}
```

Comment:
This will be verified in validation, so we can remove it from here

Reply:
Validation checks whether metrics **or** tracing endpoint is setup. 

But there might be the case where you setup tracing endpoint and then try to send metrics, which will then throw a null pointer if `metrics.endpoint` is empty.

Maybe I should just remove the validation?

Reply:
What I meant is that the `Validate` configuration method already does this check https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/709c796b5e6c3497b86551be9d652c09d2a751cb/exporter/coralogixexporter/config.go#L57-L60 both for metrics and traces. 

When the Collector is starting up it will call this `Validate` function before creating the exporter, so there is no way that either client is called with an incorrect endpoint (for either metrics or traces). Therefore we can just keep the check on `Validate` and remove the one on the client/exporter files. Does that make sense?

Reply:
Yes, but the check checks whether both `endpoint` and `metrics.endpoint` is not empty. Meaning it will pass the validation if:
```
    coralogix:
      endpoint: "tracing-ingress.coralogix.com:9443"
      metrics:
        endpoint: ""
```

But then user can try to use coralogixexporter in a metrics pipeline:
```
      metrics:
        exporters:
          - coralogix
        processors:
          - k8sattributes
          - batch
        receivers:
          - prometheus
          - otlp
```

And it will null pointer because `metrics.endpoints` is empty.

This is done because we dont want to break our users who currently use tracing pipeline.

Reply:
Oh, right, thanks for explaining this again, I see now what you meant :) There is no need to change anything then